lib/orderbook/OrderBook.ts

```diff
@@ -154,15 +163,14 @@ class OrderBook extends EventEmitter {
 
   private removePeerOrder = (orderId: string, pairId: string, decreasedQuantity?: number): boolean => {
     const matchingEngine = this.matchingEngines[pairId];
-    const ordersMap = this.peerOrders[pairId];
+    const ordersMap = this.peerOrders.get(pairId);
     if (!matchingEngine || !ordersMap) {
       this.logger.warn(`Invalid pairId: ${pairId}`);
       return false;
     }
 
-    const order = ordersMap[orderId];
-
-    if (order && matchingEngine.removePeerOrder(orderId, decreasedQuantity)) {
```

Comment:
was this causing a bug, or you just preferred to rely solely on finding the order in `MatchingEngine`?

Reply:
Yes it's a bug, I'd mentioned it:

> Fixing a bug in removePeerOrderthat was incorrectly retrieving an order from ordersMap.

`ordersMap` maps to an `Orders` type which has `buyOrders` and `sellOrders` - not order Ids - as keys. However, `MatchingEngine.removePeerOrder` returns the order we're interested in for us so that's what I used.